core/math/math_2d.cpp

```diff
@@ -62,7 +62,8 @@ Vector2 Vector2::normalized() const {
 }
 
 bool Vector2::is_normalized() const {
-	return Math::isequal_approx(length(), (real_t)1.0);
+	// use length_squared() instead of length() to avoid sqrt(), makes it more stringent.
+	return Math::is_equal_approx(length_squared(), 1.0);
```

Comment:
I'd rather keep it because it's a subtlety: for small deviations from 1.0, it effectively makes the epsilon twice as small.